src/RemoteInboxNotifications/StoredStateSetupForProducts.php

```diff
@@ -112,19 +108,23 @@ public static function run_on_transition_post_status( $new_status, $old_status,
 			return;
 		}
 
-		$stored_state                         = RemoteInboxNotificationsEngine::get_stored_state();
-		$stored_state->there_are_now_products = true;
-
-		RemoteInboxNotificationsEngine::update_stored_state( $stored_state );
-
-		self::enqueue_async_run_remote_notifications();
+		self::update_stored_state_and_run_remote_notifications();
 	}
 
 	/**
 	 * Enqueues an async action (using action-scheduler) to run remote
 	 * notifications.
 	 */
-	private static function enqueue_async_run_remote_notifications() {
+	private static function update_stored_state_and_run_remote_notifications() {
+		$stored_state = RemoteInboxNotificationsEngine::get_stored_state();
+		if ( true === $stored_state->there_are_now_products ) {
```

Comment:
Maybe also add a comment on why we're doing this check, for future context

Reply:
Added in 634f262a7f4eaa1b02311e34a3fb0f8aaf4e65f6.